Compiler errors with aggressive warnings enabled

Compiling with Clang on MacOS and compiler options `-Wall -Wextra -pedantic -Werror` produces the following error:

```
.../include/celero/ThreadLocal.h:30:9: error: keyword is hidden by macro
      definition [-Werror,-Wkeyword-macro]
```

```
clang++ version: Apple clang version 11.0.3 (clang-1103.0.32.29)
```

`thread_local` is a C++11 keyword, so it seems that the check

https://github.com/DigitalInBlue/Celero/blob/ce3b5a33c27950fe356003b7ba886e0b2cb324f1/include/celero/ThreadLocal.h#L22

will be incorrect whenever `thread_local` is not defined as a macro.

---

Compiling with G++ on MacOS and compiler options `-Wall -Wextra -pedantic -Werror` produces the following error:

```
.../src/Memory.cpp:66:19: error: 'strlen' was not declared in this scope 
   66 |    const auto i = strlen(line); 
      |
```

and many more like it.

```
g++ version: g++-9 (Homebrew GCC 9.3.0) 9.3.0
```

Looks like `Memory.cpp` is missing `#include <cstring>`.

--- a/include/celero/ThreadLocal.h
+++ b/include/celero/ThreadLocal.h
@@ -19,10 +19,8 @@
 /// limitations under the License.
 ///
 
-#ifndef thread_local
-
-#if __STDC_VERSION__ >= 201112 && !defined __STDC_NO_THREADS__
-#define thread_local _Thread_local
+#if __cplusplus >= 201103L || defined _MSC_VER && _MSC_VER >= 1900
+/* do nothing, thread_local is a c++11 keyword */
 #elif defined _WIN32 && (defined _MSC_VER || defined __ICL || defined __DMC__ || defined __BORLANDC__)
 #define thread_local __declspec(thread)
 /* note that ICC (linux) and Clang are covered by __GNUC__ */
@@ -30,7 +28,6 @@
 #define thread_local __thread
 #else
 #error "Cannot define thread_local"
-#endif
 #endif
 
 #endif

--- a/src/Memory.cpp
+++ b/src/Memory.cpp
@@ -29,6 +29,7 @@
 #include <sys/types.h>
 #include <unistd.h>
 #include <array>
+#include <cstring>
 #else
 #include <stdio.h>
 #include <stdlib.h>

